Image listing with "reference" filter raises panic

### Description

Using the `python:3` image in this example, my pod is able to run `nerdctl pull` and `nerdctl image list` just fine on the `containerd` socket mounted from the host k8s node. 

However when using `nerdctl image list --filter reference=python:3`, the command raises an exception:

```
[*]─[hostname]─[/workdir]
└──> nerdctl --address /workdir/.node/containerd.sock --namespace k8s.io image list --filter reference=python:3
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xb40cdd]

goroutine 1 [running]:
github.com/distribution/reference.FamiliarString({0x0?, 0x0?})
	/home/runner/go/pkg/mod/github.com/distribution/reference@v0.6.0/helpers.go:31 +0x5d
github.com/distribution/reference.FamiliarMatch({0xc0003e43fa, 0x8}, {0x0, 0x0})
	/home/runner/go/pkg/mod/github.com/distribution/reference@v0.6.0/helpers.go:37 +0x31
github.com/containerd/nerdctl/v2/pkg/referenceutil.(*ImageReference).FamiliarMatch(...)
	/home/runner/work/nerdctl/nerdctl/pkg/referenceutil/referenceutil.go:68
github.com/containerd/nerdctl/v2/pkg/imgutil.matchesReferences({{0xc000124eb0, 0x47}, 0xc0003ce3f0, {{0xc0005021c0, 0x39}, {0xc000124f00, 0x47}, 0x2e7, {0x0, 0x0, ...}, ...}, ...}, ...)
	/home/runner/work/nerdctl/nerdctl/pkg/imgutil/filtering.go:334 +0xad
github.com/containerd/nerdctl/v2/pkg/cmd/image.List.FilterByReference.func5.1({{0xc000124eb0, 0x47}, 0xc0003ce3f0, {{0xc0005021c0, 0x39}, {0xc000124f00, 0x47}, 0x2e7, {0x0, 0x0, ...}, ...}, ...})
	/home/runner/work/nerdctl/nerdctl/pkg/imgutil/filtering.go:269 +0x58
github.com/containerd/nerdctl/v2/pkg/imgutil.filter[...]({0xc00054e008, 0x35, 0x1120f40}, 0xc000176800?)
	/home/runner/work/nerdctl/nerdctl/pkg/imgutil/filtering.go:295 +0x102
github.com/containerd/nerdctl/v2/pkg/cmd/image.List.FilterByReference.func5({0xc00054e008, 0x35, 0x55})
	/home/runner/work/nerdctl/nerdctl/pkg/imgutil/filtering.go:268 +0xa5
github.com/containerd/nerdctl/v2/pkg/imgutil.ApplyFilters({0xc00054e008?, 0x1?, 0xc000528190?}, {0xc00028c088, 0x1, 0x0?})
	/home/runner/work/nerdctl/nerdctl/pkg/imgutil/filtering.go:129 +0x4b
github.com/containerd/nerdctl/v2/pkg/cmd/image.List({0x14b00b8, 0xc000528190}, 0xc00002d600, {0xc00005b740, 0x1, 0x1}, {0x0, 0x0, 0x0})
	/home/runner/work/nerdctl/nerdctl/pkg/cmd/image/list.go:104 +0x605
github.com/containerd/nerdctl/v2/pkg/cmd/image.ListCommandHandler({0x14b00b8, 0xc000528190}, 0xc00002d600, 0xc000454640)
	/home/runner/work/nerdctl/nerdctl/pkg/cmd/image/list.go:53 +0x58
github.com/containerd/nerdctl/v2/cmd/nerdctl/image.imagesAction(0xc00043b208, {0xc0003a13e0?, 0x0?, 0x2?})
	/home/runner/work/nerdctl/nerdctl/cmd/nerdctl/image/image_list.go:143 +0xd0
github.com/spf13/cobra.(*Command).execute(0xc00043b208, {0xc00014a070, 0x2, 0x2})
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:985 +0xaaa
github.com/spf13/cobra.(*Command).ExecuteC(0xc000458008)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1117 +0x3ff
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.8.1/command.go:1041
main.xmain()
	/home/runner/work/nerdctl/nerdctl/cmd/nerdctl/main.go:145 +0x8f
main.main()
	/home/runner/work/nerdctl/nerdctl/cmd/nerdctl/main.go:128 +0x13
```


**What could be causing this?**

### Steps to reproduce the issue

1. Pull the `python:3` image:

```
[*]─[hostname]─[/workdir]
└──> nerdctl --address /workdir/.node/containerd.sock --namespace k8s.io pull python:3
docker.io/library/python:3:                                                       resolved       |++++++++++++++++++++++++++++++++++++++|
index-sha256:385ccb8304f6330738a6d9e6fa0bd7608e006da7e15bc52b33b0398e1ba4a15b:    exists         |++++++++++++++++++++++++++++++++++++++|
manifest-sha256:be98a0e94e3f445806ddc30a8d5a962cc2ac807f44de4d9aad4a8ea3664746b6: exists         |++++++++++++++++++++++++++++++++++++++|
config-sha256:36d17f72f4f3630dbc93a0b2fe6f7e61179c597cbef32f815b34f70ef1e402f2:   exists         |++++++++++++++++++++++++++++++++++++++|
elapsed: 1.4 s                                                                    total:   0.0 B (0.0 B/s)
```

2. Verify that listing the images with no filter shows the expected image:

```
[x]─[hostname]─[/workdir]
└──> nerdctl --address /workdir/.node/containerd.sock --namespace k8s.io image list | grep "python"
python                                                                               3                             385ccb8304f6    23 seconds ago    linux/amd64    1.085GB    381.8MB
```

3. Try using `--filter reference=python:3` and see that it raises a runtime error (nil pointer dereference):

```
[*]─[hostname]─[/workdir]
└──> nerdctl --address /workdir/.node/containerd.sock --namespace k8s.io image list --filter reference=python:3
panic: runtime error: invalid memory address or nil pointer dereference
...
```

### Describe the results you received and expected

Saw an internal panic where i expected to see the image i filtered by `reference`

### What version of nerdctl are you using?

```
[x]─[hostname]─[/workdir]
└──> nerdctl --version
nerdctl version 2.0.3
```

### Are you using a variant of nerdctl? (e.g., Rancher Desktop)

None

### Host information

Within a k8s pod on EKS EC2 node

--- a/hack/build-integration-kubernetes.sh
+++ b/hack/build-integration-kubernetes.sh
@@ -50,7 +50,8 @@ install::kubectl(){
   local temp
   temp="$(fs::mktemp "install")"
 
-  http::get "$temp"/kubectl "https://storage.googleapis.com/kubernetes-release/release/$version/bin/linux/${GOARCH:-amd64}/kubectl"
+  http::get "$temp"/kubectl "https://dl.k8s.io/release/$version/bin/linux/${GOARCH:-amd64}/kubectl"
+
   host::install "$temp"/kubectl
 }
 

--- a/pkg/referenceutil/referenceutil.go
+++ b/pkg/referenceutil/referenceutil.go
@@ -65,7 +65,10 @@ func (ir *ImageReference) FamiliarName() string {
 }
 
 func (ir *ImageReference) FamiliarMatch(pattern string) (bool, error) {
-	return reference.FamiliarMatch(pattern, ir.nn)
+	if ir.nn != nil {
+		return reference.FamiliarMatch(pattern, ir.nn)
+	}
+	return false, nil
 }
 
 func (ir *ImageReference) String() string {

